src/opm/output/eclipse/WriteInit.cpp

```diff
@@ -559,6 +558,63 @@ namespace {
 
         initFile.write("TRANNNC", singlePrecision(tran));
     }
+
+    // output aquifer cell and aquifer connection information for numerical aquifers
+    void writeNumericalAquifers(const Opm::NumericalAquifers& num_aquifers,
+                                const ::Opm::EclipseGrid&          grid,
+                                ::Opm::EclIO::OutputStream::Init&  initFile)
+    {
+        std::vector<int> aquifern(grid.getNumActive(), 0);
+        // aquifer cells
+        const auto& aquifer_cells = num_aquifers.allAquiferCells();
+        for ([[maybe_unused]] const auto& [cell_idx, cell] : aquifer_cells) {
+            const size_t active_index = grid.activeIndex(cell->global_index);
+            aquifern[active_index] -= std::exp2(cell->aquifer_id - 1);
+        }
+
+        // aquifer connections
+        for (const auto& [id, aqu] : num_aquifers.aquifers()) {
+            const auto& connections = aqu.connections();
+            const int exp2_id_1 = std::exp2(id - 1);
```

Comment:
Note that `std::exp2()` returns a floating-point result.  Is there a reason for not using bitshifts here?
```C++
const int exp2_id_1 = 1 << (id - 1);
```

Reply:
I seldom use bitwise operation, so it was not in my mind. I will change to it. 

And also I knew `std::exp2()` return floating-point results, while I think it is pretty safe for this case.